File: FCNNF/FCNNF/otherDocument/MinstHWDataLoading.cpp
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#define MAX_LINE 16

float absolute(float a)
{
	if (a < 0)
	{
		return -a;
	}
	else
	{
		return a;
	}
}

char* F2S(float f, char* str)
{
	char str1[40];
	int j = 0, k, i;
	float d;

	d = absolute(f);
	i = (int)d;//浮点数的整数部分
	while (i > 0)
	{
		str1[j++] = i % 10 + '0';
		i = i / 10;
	}
	for (k = 0; k < j; k++)
	{
		str[k] = str1[j - 1 - k];//被提取的整数部分正序存放到另一个数组
	}

	if ((int)d == 0)
	{
		str[j++] = '0';
	}

	str[j++] = '.';

	d = d - (int)d;//小数部分提取
				   //printf("小数部分%f\n", d);
	for (i = 0; i < 4; i++)
	{
		d = d * 10;
		str[j++] = (int)d + '0';
		//printf("%c", str[j - 1]);
		d = d - (int)d;

	}
	/*while (str[--j] == '0');*/
	str[j] = '\0';

	/*printf("%c\n", str[0]);*/

	//处理负数
	if (f < 0)
	{
		j = 0;
		while (str[j] != '\0')
		{
			++j;
		}
		str[j + 1] = '\0';
		while (j > 0)
		{
			str[j] = str[j - 1];
			--j;
		}
		str[0] = '-';
	}
	return str;
}

struct _iobuf {
	char *_ptr;
	int   _cnt;
	char *_base;
	int   _flag;
	int   _file;
	int   _charbuf;
	int   _bufsiz;
	char *_tmpfname;
};
typedef struct _iobuf FILE;

float nums[54880000] = { 0 };
float labels[70000] = { 0 };

int main()
{
	char buf[MAX_LINE];  /*缓冲区*/
	FILE *fp = NULL;     /*文件指针*/
	FILE *fp2 = NULL;
	int len;             /*行字符个数*/
	int i = 0;
	int j = 0;
	char * flag;

	//if ((fp = fopen("C:/Users/Administrator/Documents/GitHub/MinstHW/DataFeatrue.txt", "r")) == NULL)
	//{
	//	perror("fail to read");
	//	return 0;
	//}

	fp = fopen("C:/Users/Administrator/Documents/GitHub/MinstHW/DataFeatrue.txt", "r");
	flag = fgets(buf, MAX_LINE, fp);
	while (flag != NULL)
	{
		len = strlen(buf);
		buf[len - 1] = '\0';  /*去掉换行符*/
		//printf("%s %d \n", buf, len - 1);
		nums[i++] = atof(buf);
		flag = fgets(buf, MAX_LINE, fp);
	}
	fclose(fp);

	//for (i = 0; i < 54880000; ++i) {
	//	printf("%d:  %s\n", i, F2S(nums[i], buf));
	//}



	// labels
	fp2 = fopen("C:/Users/Administrator/Documents/GitHub/MinstHW/DataLabel.txt", "r");
	flag = fgets(buf, MAX_LINE, fp2);
	while (flag != NULL)
	{
		len = strlen(buf);
		buf[len - 1] = '\0';  /*去掉换行符*/
		//printf("%s %d \n", buf, len - 1);
		labels[j++] = atof(buf);
		flag = fgets(buf, MAX_LINE, fp2);
	}
	fclose(fp2);

	//for (i = 0; i < 70000; ++i) {
	//	printf("%d:  %s\n", i, F2S(labels[i], buf));
	//}
	return 0;
}

File: MC8.0/MC8.0/MinstHWdataloading.m
frame(nums,labels) and (
 function absolute ( float a,float RValue )
 {
     frame(return) and ( 
     int return<==0 and skip;
     if(a<0) then 
     {
         return<==1 and RValue:=-a;
         skip
     }
     else
     {
         return<==1 and RValue:=a;
         skip
     }
     )
     }; 
  function F2S ( float f,char *str,char* RValue )
 {
     frame(F2S_str1,F2S_j,F2S_k,F2S_i,F2S_d,return) and ( 
     int return<==0 and skip;
     char F2S_str1[40] and skip;
     int F2S_j<==0,F2S_k,F2S_i and skip;
     float F2S_d and skip;
     F2S_d:=absolute(f,RValue);
     F2S_i:=(int)F2S_d;
     while( (F2S_i>0) )
     {
         F2S_str1[F2S_j]:=F2S_i % 10+'0';
         F2S_j:=F2S_j+1;
         F2S_i:=F2S_i/ 10
     };
     F2S_k:=0;
     
     while( (F2S_k<F2S_j) )
     {
         str[F2S_k]:=F2S_str1[F2S_j-1-F2S_k];
         F2S_k:=F2S_k+1
         
     };
     if((int)F2S_d=0) then 
     {
         str[F2S_j]:='0';
         F2S_j:=F2S_j+1
         
     }
     else 
     {
          skip 
     };
     str[F2S_j]:='.';
     F2S_j:=F2S_j+1;
     F2S_d:=F2S_d-(int)F2S_d;
     F2S_i:=0;
     
     while( (F2S_i<4) )
     {
         F2S_d:=F2S_d*10;
         str[F2S_j]:=(int)F2S_d+'0';
         F2S_j:=F2S_j+1;
         F2S_d:=F2S_d-(int)F2S_d;
         F2S_i:=F2S_i+1
         
     };
     str[F2S_j]:='\0';
     if(f<0) then 
     {
         F2S_j:=0;
         while( (str[F2S_j]!='\0') )
         {
             F2S_j:=F2S_j+1
         };
         str[F2S_j+1]:='\0';
         while( (F2S_j>0) )
         {
             str[F2S_j]:=str[F2S_j-1];
             F2S_j:=F2S_j-1
         };
         str[0]:='-'
         
     }
     else 
     {
          skip 
     };
     return<==1 and RValue:=str;
     skip
     )
     }; 
      struct _iobuf {
     char *_ptr and 
     int _cnt and 
     char *_base and 
     int _flag and 
     int _file and 
     int _charbuf and 
     int _bufsiz and 
     char *_tmpfname 
 };
 float nums[54880000]<=={0} and skip;
 float labels[70000]<=={0} and skip;
 function main ( int  RValue )
 {
     frame(main_buf,main_fp,main_fp2,main_len$,main_i,main_j,main_flag,return) and (
     int return<==0 and skip;
     char main_buf[16] and skip;
     FILE *main_fp<==NULL and skip;
     FILE *main_fp2<==NULL and skip;
     int main_len$ and skip;
     int main_i<==0 and skip;
     int main_j<==0 and skip;
     char *main_flag and skip;
     main_fp:=fopen("C:/Users/Administrator/Documents/GitHub/MinstHW/DataFeatrue.txt","r");
     main_flag:=fgets(main_buf,16,main_fp);
     while( (main_flag!=NULL) )
     {
         main_len$:=strlen(main_buf);
         main_buf[main_len$-1]:='\0';
         nums[main_i]:=atof(main_buf);
         main_i:=main_i+1;
         main_flag:=fgets(main_buf,16,main_fp)
     };
     fclose(main_fp) and skip;
     main_fp2:=fopen("C:/Users/Administrator/Documents/GitHub/MinstHW/DataLabel.txt","r");
     main_flag:=fgets(main_buf,16,main_fp2);
     while( (main_flag!=NULL) )
     {
         main_len$:=strlen(main_buf);
         main_buf[main_len$-1]:='\0';
         labels[main_j]:=atof(main_buf);
         main_j:=main_j+1;
         main_flag:=fgets(main_buf,16,main_fp2)
     };
     fclose(main_fp2) and skip;
     return<==1 and RValue:=0;
     skip
     )
 };
  main(RValue)
 )
